frontend/msSettingsManager.pas

```diff
@@ -480,6 +482,37 @@ procedure TSettingsManager.BuildFromPluginsItemClick(Sender: TObject);
   slSelection.Free;
 end;
 
+procedure TSettingsManager.SelectSimilarNodesItemClick(Sender: TObject);
+var
+  i, index: Integer;
+  tvNode: TTreeNode;
+  reqSmashType, currentSmashType: TSmashtype;
+  slSelection: TStringList;
+begin
+  slSelection := TStringList.Create;
+  try
+    for i := 0 to Pred(tvRecords.SelectionCount) do
+    begin
+      tvNode := tvRecords.Selections[i];
+      reqSmashType := TElementData(tvNode.Data).smashType;
+      slSelection.AddObject(tvNode.Text, TObject(reqSmashType));
+    end;
+    for i := 0 to Pred(tvRecords.Items.Count) do begin
+      currentSmashType := TElementData(tvRecords.Items[i].Data).smashType;
+      index := slSelection.IndexOf(tvRecords.Items[i].Text);
+      if (index > -1) then
+      begin
+        reqSmashType := TSmashType(slSelection.Objects[index]);
+          if (currentSmashType = reqSmashType) and (tvRecords.Items[i].Text = slSelection[index]) then
+            if tvRecords.Items[i].Selected <> True then
```

Comment:
You don't need to do a `<>` comparison with `true`, just append `not`.  So this line would be:

`if not tvRecords.Items[i].Selected then`